### PR TITLE
Replace nan with infinity

### PR DESCRIPTION
**Pull request by imakhlin**
- replace NaN in functions and aggregators by math.Inf(...)
- update init_array - use default aggregator's value

### CODE REVIEW

File: pkg/aggregate/functions.go [hunk 1]
```diff
@@ -114,10 +115,10 @@ func (a *MinAggregator) UpdateExpr(col string, bucket int) string {
 // Maximum Aggregator
 type MaxAggregator struct{ FloatAggregator }
 
-func (a *MaxAggregator) Clear() { a.val = -math.MaxFloat64 } // TODO: use math.Inf(-1)
+func (a *MaxAggregator) Clear() { a.val = math.Inf(-1) }
 
 func (a *MaxAggregator) Aggregate(t int64, v float64) {
-	if !math.IsNaN(v) && (math.IsNaN(a.val) || v > a.val) {
+	if utils.IsDefined(v) && (utils.IsUndefined(a.val) || v > a.val) {
```
Comment (gtopper) [inline pkg/aggregate/functions.go]: You shouldn't have to defend from ±Inf - it should just work.

File: pkg/aggregate/functions.go [hunk 2]
```diff
@@ -91,18 +92,18 @@ func (a *SumAggregator) Aggregate(t int64, v float64) {
 type SqrAggregator struct{ FloatAggregator }
 
 func (a *SqrAggregator) Aggregate(t int64, v float64) {
-	if !math.IsNaN(v) {
+	if utils.IsDefined(v) {
 		a.val += v * v
```
Comment (gtopper) [inline pkg/aggregate/functions.go]: Where would these nonnumerical values come from?

File: pkg/aggregate/functions.go [hunk 3]
```diff
@@ -75,14 +75,15 @@ func (a *FloatAggregator) UpdateExpr(col string, bucket int) string {
 }
 
 func (a *FloatAggregator) InitExpr(col string, buckets int) string {
-	return fmt.Sprintf("_%s_%s=init_array(%d,'double');", col, a.attr, buckets)
+	// TODO: consider adding default init value per aggregator
+	return fmt.Sprintf("_%s_%s=init_array(%d,'double', %f);", col, a.attr, buckets, a.val)
```
Comment (gtopper) [inline pkg/aggregate/functions.go]: Let's be consistent on spacing.

File: pkg/aggregate/aggregate.go [hunk 4]
```diff
@@ -132,13 +132,13 @@ func NewAggregatorList(aggrType AggrType) *AggregatorList {
...
-		list = append(list, &MinAggregator{FloatAggregator{attr: "min", val: math.MaxFloat64}}) // TODO: use math.Inf(1)
+		list = append(list, &MinAggregator{FloatAggregator{attr: "min", val: math.Inf(1)}})
 	}
 	if (aggrType & aggrTypeMax) != 0 {
-		list = append(list, &MaxAggregator{FloatAggregator{attr: "max", val: -math.MaxFloat64}}) // TODO: use math.Inf(-1)
+		list = append(list, &MaxAggregator{FloatAggregator{attr: "max", val: math.Inf(-1)}})
 	}
 	if (aggrType & aggrTypeLast) != 0 {
-		list = append(list, &LastAggregator{FloatAggregator{attr: "last"}, 0})
+		list = append(list, &LastAggregator{FloatAggregator{attr: "last", val: math.Inf(-1)}, 0})
```
Comment (gtopper) [inline pkg/aggregate/aggregate.go]: I don't think -Inf is really appropriate here. It shouldn't matter what we set it to, but still.

File: pkg/aggregate/functions.go (hunk 3 above)
Comment (imakhlin) [reply pkg/aggregate/functions.go]: it's `go fmt ../../...` I have nothing to do with that

File: pkg/aggregate/functions.go (hunk 2 above)
Comment (imakhlin) [reply pkg/aggregate/functions.go]: User input could contain NaN for example

File: pkg/aggregate/aggregate.go (hunk 4 above)
Comment (imakhlin) [reply pkg/aggregate/aggregate.go]: Just using the same value as have been used at initialisation: `&LastAggregator{FloatAggregator{attr: "last", val: math.Inf(-1)}`

File: pkg/aggregate/functions.go (hunk 2 above)
Comment (imakhlin) [reply pkg/aggregate/functions.go]: User input... NaN in CSV file for example

Comment (gtopper) [reply pkg/aggregate/functions.go]: Sure. But then it could also be Inf or -Inf, and it's incorrect to ignore those.

Comment (gtopper) [reply pkg/aggregate/functions.go]: Also, it would be legitimate for us to reject nonnumericals as input.

File: pkg/aggregate/functions.go (hunk 3 above)
Comment (gtopper) [reply pkg/aggregate/functions.go]: I mean spacing within the update expression - that's not `go fmt`'s fault :)

File: pkg/aggregate/aggregate.go (hunk 4 above)
Comment (gtopper) [reply pkg/aggregate/aggregate.go]: Still makes no sense, though.

File: pkg/aggregate/functions.go (hunk 2 above)
Comment (imakhlin) [reply pkg/aggregate/functions.go]: Agree on rejecting invalid input. 
Regarding the +Inf/-Inf I see no reason to threat them as numeric values

File: pkg/aggregate/functions.go (hunk 3 above)
Comment (imakhlin) [reply pkg/aggregate/functions.go]: it so negligible ...

**Pull request merged**